### ISSUE TITLE
Fix warning "module.parent.require is not supported by webpack"

### ISSUE THREAD

**Issue by tansongyang** [MindscapeHQ/raygun4node#138]
I have a repro of this issue at https://github.com/tansongyang/raygun-webpack-issue-repro.

Run `npm run build` in the console. Here's the output:

```
> next build

   ▲ Next.js 14.1.0

   Creating an optimized production build ...
 ⚠ Compiled with warnings

./node_modules/raygun/build/raygun.js
module.parent.require is not supported by webpack.

Import trace for requested module:
./node_modules/raygun/build/raygun.js
./app/raygun.ts
./app/page.tsx

./node_modules/raygun/build/raygun.js
Module not found: Can't resolve 'raygun-apm/lib/src/crash_reporting' in '/Users/frank/Code/raygun-webpack-issue-repro/node_modules/raygun/build'

Import trace for requested module:
./node_modules/raygun/build/raygun.js
./app/raygun.ts
./app/page.tsx
```

Source appears to be here: https://github.com/MindscapeHQ/raygun4node/blob/v0.13.0/lib/raygun.ts#L51

Not sure what the fix is, but if I navigate to the file in VS Code, I get a warning that `module.parent` is deprecated:

![image](https://github.com/MindscapeHQ/raygun4node/assets/9488719/4fb521c2-00dd-46fd-a090-c48b76fa92b9)

**Comment by rob-mcgrail** [comment]
We've got the same issue

**Comment by miquelbeltran** [comment]
Hello, sorry we took so long to respond here. I am now taking care of keeping this project up to date.
Thanks a lot for the sample project with the issue, that will help me debug and fix.
I'll see what I can do today.

**Comment by miquelbeltran** [comment]
After looking into it, the main issue is how the package attempts to load the `raygun-apm` dependency.
As it tries to load it, and if fails. This is wrapped by a try-catch block, but still causes warnings to appear.

This causes two warnings when building the app:

```
1. module.parent.require is not supported by webpack.
2. Module not found: Can't resolve 'raygun-apm/lib/src/crash_reporting' in '...
```

I think a proper solution would be to change the way we provide the `apmBridge` internally, instead of trying to attempt to load it, I think we should expose a setter and let users provide it externally rather than attempt to load in internally.

I'll have to consult with the rest of the team on how to move forward with this.

**Comment by miquelbeltran** [comment]
With the changes in the PR #222 the build output should be:

```
➜  raygun-webpack-issue-repro git:(main) ✗ npm run build

> raygun-webpack-issue-repro@0.1.0 build
> next build

   ▲ Next.js 14.1.0

   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
 ✓ Collecting page data    
   Generating static pages (0/5)  [    ]paste_your_api_key_here
paste_your_api_key_here
 ✓ Generating static pages (5/5) 
 ✓ Collecting build traces    
 ✓ Finalizing page optimization    

Route (app)                              Size     First Load JS
┌ ○ /                                    5.14 kB        89.3 kB
└ ○ /_not-found                          882 B            85 kB
+ First Load JS shared by all            84.1 kB
  ├ chunks/69-1b6d135f94ac0e36.js        28.9 kB
  ├ chunks/fd9d1056-cc48c28d170fddc2.js  53.4 kB
  └ other shared chunks (total)          1.87 kB


○  (Static)  prerendered as static content
```

**Comment by lance-h** [comment]
I can't see this fix in the latest version, 2.1.2. Is there any indication of when we can get our hands on this?

**Comment by miquelbeltran** [comment]
The solution for this issue was reverted in 1.0.0 because it broke integration with Raygun APM.

There are currently no plans to fix this compilation warning.  

A note was added to the package README: https://github.com/MindscapeHQ/raygun4node/blob/develop/README.md#known-issues
